src/test/common/steps/navigation.js

```diff
@@ -0,0 +1,88 @@
+const { Given, When } = require('cucumber')
+
+const pages = require('./pages')
+const { openOverviewPage, enterNameAndSubmit, enterNinoAndSubmit, enterDateOfBirth, selectNoOnPregnancyPage, completeTheApplicationAsAWomanWhoIsNotPregnant } = require('./common-steps')
+const { VALID_NINO, FIRST_NAME, LAST_NAME, DAY, MONTH, YEAR } = require('./constants')
+
+const ENTER_NAME_PAGE = 'enter name'
+const ENTER_NINO_PAGE = 'enter national insurance'
+const ENTER_DOB_PAGE = 'enter date of birth'
+const ARE_YOU_PREGNANT_PAGE = 'are you pregnant'
+const CARD_ADDRESS_PAGE = 'card address'
+const OVERVIEW_PAGE = 'HTBHF overview'
+const CHECK_PAGE = 'check details'
+
+Given(/^I have entered my details up to the (.*) page$/, async function (page) {
+  await enterDetailsUpToPage(page)
+})
+
+When(/^I click the Cookies link$/, async function () {
+  await pages.overview.clickCookieLink()
+})
+
+When(/^I navigate to the (.*) page$/, async function (page) {
+  await navigateToPage(page)
+})
+
+async function navigateToPage (page) {
+  const baseUrl = pages.url
+  switch (page) {
```

Comment:
The alternative to a switch statement would be to map the `page` to a function - but not sure if its any better:
```
const pageMap = {
  [ENTER_NAME_PAGE]: (baseUrl) => await pages.enterName.openEnterNameUrl(baseUrl),
  [ENTER_NINO_PAGE]: (baseUrl) => pages.enterNino.openEnterNinoUrl(baseUrl)
  ...
}

...

try {
  pageMap[page](baseUrl)
} catch (error) {
  throw new Error(...)
}
```